src/ui/components/MainPage.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react';
+
+const MainPage: React.FC = () => {
+  const [jsonInput, setJsonInput] = useState('');
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
```

Comment:
This needs to use React.useCallback or it will cause a re-render on every character typed.